server/lib/job-queue/handlers/move-to-object-storage.ts

```diff
@@ -0,0 +1,138 @@
+import * as Bull from 'bull'
+import { logger } from '@server/helpers/logger'
+import { MoveObjectStoragePayload } from '../../../../shared'
+import { VideoModel } from '@server/models/video/video'
+import { generateObjectStoreUrl, storeObject } from '@server/lib/object-storage'
+import { CONFIG } from '@server/initializers/config'
+import { join } from 'path'
+import { HLS_STREAMING_PLAYLIST_DIRECTORY } from '@server/initializers/constants'
+import { getHlsResolutionPlaylistFilename } from '@server/lib/video-paths'
+import { MVideoWithAllFiles, VideoStorageType } from '@server/types/models'
+import { remove } from 'fs-extra'
+import { publishAndFederateIfNeeded } from '@server/lib/video'
+import { VideoJobInfoModel } from '@server/models/video/video-job-info'
+
+export async function processMoveToObjectStorage (job: Bull.Job) {
+  const payload = job.data as MoveObjectStoragePayload
+  logger.info('Moving video %s in job %d.', payload.videoUUID, job.id)
+
+  const video = await VideoModel.loadWithFiles(payload.videoUUID)
+  // No video, maybe deleted?
+  if (!video) {
+    logger.info('Can\'t process job %d, video does not exist.', job.id)
+    return undefined
+  }
+
+  if (CONFIG.TRANSCODING.WEBTORRENT.ENABLED && video.VideoFiles) {
+    await moveWebTorrentFiles(video, payload.videoFileId)
```

Comment:
If transcoding is disabled we use webtorrent files by default so you need to call `moveWebTorrentFiles`